taskfile/node.go

```diff
@@ -18,6 +18,7 @@ type Node interface {
 	Read(ctx context.Context) ([]byte, error)
 	Parent() Node
 	Location() string
+	RedactedLocation() string
```

Comment:
Just had a quick look on my phone, not a full review.

Is the raw (non-redacted) location actually used outside of the git node implementation? You could make the argument that the credentials should never be exposed via the interface and that `Location` should just return the redacted version. This would save us having to add an interface method that only one of the implementations needs.

Reply:
I agree with you @pd93 !
AFAIK, the `Location` is used only to reference it in the `graph` so, imho no objection to return the `redacted` version.
Also, the http node implementation could / should have the same behavior, as we can use `http://USER:PASS@URL` pattern

Reply:
That would make the whole change a lot simpler! I saw that the `node.Location` is used to provide the cache key.

I think ignoring the username and password combination for the cache should be fine since different combinations shouldn't yield different Taskfile contents.

Adding the HTTP node redaction should be easy too.